Tests/SwiftSyntaxTest/SyntaxTests.swift

```diff
@@ -168,4 +168,50 @@ class SyntaxTests: XCTestCase {
     XCTAssertEqual(funcKeywordInTree2?.as(TokenSyntax.self)?.tokenKind, .keyword(.func))
     XCTAssertNotEqual(funcKeywordInTree1.id, funcKeywordInTree2?.id)
   }
+
+  func testIntegerLiteralExprSyntax() {
+    let testCases: [UInt: (String, Int?)] = [
+      #line: ("2", 2),
+      #line: ("02", 2),
+      #line: ("020", 20),
+      #line: ("-02", -2),
```

Comment:
Should cases like `-0xA`, `-0o7`, and `-0b1` also be handled? I should note though that we parse the `-` as an operator, so these cases at least won't come for parsed source code.

Reply:
@hamishknight I've added the test cases. 
Are the results what you expect `nil`?

Reply:
I don’t think we need to care about negative cases. Negative numbers should be represented by a number literal with a  prefix operator in the syntax tree.

Reply:
In that case, yeah LGTM